### DISCUSSION TITLE
To hold the left clicking for drag and drop​

### DISCUSSION THREAD

**Discussion by tangle5500** [pit-ray/win-vind#218] [General]

**Comment by pit-ray** [comment]
I'll reply in English because I cannot write Chinese.

To hold the left clicking for drag and drop, you can use GUI visual mode.
https://pit-ray.github.io/win-vind/cheat_sheet/functions/#to_gui_visual

The mapping of the GUI visual is only supported in the big tier or huge tier by default, but you can also use this feature in the smaller tier.
https://pit-ray.github.io/win-vind/cheat_sheet/defaults/

In the end, if this discussion is still active, please set the proper title in English.
Thanks.

**Reply by tangle5500** [reply]
好的，谢谢您，我还有其它问题想请教您

1. 鼠标不可以斜向移动，只可以横向纵向移动吗？比如同时按下hj向左下移动

2. 插入模式这种映射不起作用，在其它模式是可以的

```
inoremap <ctrl-1> <up>
inoremap <ctrl-2> <to_edi_normal>:!cmd<cr>
```

3. 是否有块可视化模式，我只看到了字符和行可视化模式

**Reply by pit-ray** [reply]
1. The first version of win-vind supported diagonal moving, but now that it is not supported to respect for Vim.
  Instead of that, you can use [easyclick](https://pit-ray.github.io/win-vind/cheat_sheet/functions/#easyclick) or [gridmove](https://pit-ray.github.io/win-vind/cheat_sheet/functions/#gridmove) for complex moving.

2. Please try `imap` instead of `inoremap`, because `inoremap` only maps to the default mapping.

3. What is a block? For text, the [blockstylecaret](https://pit-ray.github.io/win-vind/cheat_sheet/options/#block-style-caret) is available.

Thanks.

**Reply by tangle5500** [reply]
1. [easyclick](https://pit-ray.github.io/win-vind/cheat_sheet/functions/#easyclick) 和 [gridmove](https://pit-ray.github.io/win-vind/cheat_sheet/functions/#gridmove) 我同时都在用，而且非常棒，如果能斜向移动就更好了，既然现在不支持了，我知道了，那就没有问题了

2.  我一开始使用的就是 `imap` 但它和 `inoremap` 是一样的结果。`inoremap <ctrl-2> <to_edi_normal>:!cmd<cr>` 它切换了模式也出现了`:`，就好像后面 `!cmd<cr>` 没有运行下去。`<ctrl-1>`这个映射是不管是 `inoremap <ctrl-1> <up>` 还是 `imap <ctrl-1> <up>` 它都没有向上动，只有在插入模式下有问题。如果是 `map <ctrl-1> <up>` 在其它模式下没有问题都会向上动

3. 块模式就是如图这样，它可以编辑一块字符增删改，vim 块可视化模式 linux 下是 `ctrl + v`。如果您说的 [blockstylecaret ](https://pit-ray.github.io/win-vind/cheat_sheet/options/#block-style-caret) 是一样的功能，能说一下怎么用吗，我看了一下，但还是不知道怎样去用
![2023-12-04_11-21-21](https://github.com/pit-ray/win-vind/assets/55052302/e2414ea2-21bc-407e-9cb4-e4d9a79456eb)

Thank you

**Reply by pit-ray** [reply]
I didn't notice the notification. Sorry for my late reply.

2. Please try `imap <ctrl-1> <to_command>!cmd<cr><to_edi_normal>`
3.  WinAPI cannot achieve the block such as `<c-v>`. 

Thanks.
